### DISCUSSION TITLE
Styling text inside query builder dropdown

### DISCUSSION THREAD

**Discussion by kamelabiad** [react-querybuilder/react-querybuilder#570] [Q&A]
Is it possible to apply styling for the text inside the query builder dropdown ?

**Comment by jakeboone02** [comment]
The default dropdown components just render standard HTML5 `<select>` elements with some classes added, but no additional styling. You can style the RQB dropdown components in any way you can style a regular `<select>`. You can also replace the dropdown components completely with your own component.

Relevant documentation:

- [Styling](https://react-querybuilder.js.org/docs/category/styling)
- [Custom components](https://react-querybuilder.js.org/docs/tips/custom-with-fallback)

**Reply by kamelabiad** [reply]
![screenshot](https://github.com/react-querybuilder/react-querybuilder/assets/9218860/557e2b35-99ba-4512-8883-993323fc1cbf)
I have followed the above links, but i have noticed that the values dropdown is changing and not the field name , in my case i want to update the first dropdown which Contains "Dte range,DOB,Name" and use my own component

**Reply by jakeboone02** [reply]
Can you provide some example code? A working sandbox would be best. To use a custom component for the field dropdown you need to assign the component with `controlElements={{ fieldSelector: YourComponentName }}`.

**Reply by kamelabiad** [reply]
Hi again , this is my code

```tsx
import React from "react";
import FormControl from "@mui/material/FormControl";
import NativeSelect from "@mui/material/NativeSelect";

const CustomDropdown = (props) => {
  return (
    <FormControl style={{ width: "50%" }}>
      <NativeSelect
        defaultValue={props.value}
        onChange={(event) => {
          props.handleOnChange(event.target.value);
        }}
      >
        {props.options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </NativeSelect>
    </FormControl>
  );
};

export default CustomDropdown;
```

```tsx
                <QueryBuilderAntD>
                  <QueryBuilder
                    query={query}
                    fields={ruleFields.fields}
                    onQueryChange={handleQueryChange}
                    controlElements={{ fieldSelector: CustomDropdown }}
                  />
                </QueryBuilderAntD>
```

"For some reason, when using this custom dropdown, the value field type is not changing based on my selection. It used to work for textboxes, radio buttons, and checkboxes, but after implementing this approach, it consistently displays as a textbox."
![image](https://github.com/react-querybuilder/react-querybuilder/assets/9218860/9f6d3093-c799-4f59-aba9-6580ffac4d7d)

**Reply by jakeboone02** [reply]
At first glance, I can't see an issue with your code as far as functionality, but I think part of what's wrong with what you're seeing is that your base compatibility provider is Ant Design-based (`<QueryBuilderAntD />`) while your custom dropdown is MUI-based (`import ... from '@mui/material'`). My guess it that's unintentional, and you should use `<QueryBuilderMaterial />` instead of `<QueryBuilderAntD />`.
